parser/src/lib.rs

```diff
@@ -537,207 +537,55 @@ pub fn parse(format_string: &str, mode: ParserMode) -> Result<Vec<Fragment<'_>>,
 #[cfg(test)]
 mod tests {
     use super::*;
-
-    #[test]
-    fn all_parse_param_cases() {
-        // no `Param` field present - 1 case
-        assert_eq!(
-            parse_param("", ParserMode::Strict),
-            Ok(Param {
-                index: None,
-                ty: Type::Format,
-                hint: None,
-            })
-        );
-
-        // only one `Param` field present - 3 cases
-        assert_eq!(
-            parse_param("=u8", ParserMode::Strict),
-            Ok(Param {
-                index: None,
-                ty: Type::U8,
-                hint: None,
-            })
-        );
-
-        assert_eq!(
-            parse_param(":a", ParserMode::Strict),
-            Ok(Param {
-                index: None,
-                ty: Type::Format,
-                hint: Some(DisplayHint::Ascii),
-            })
-        );
-
-        assert_eq!(
-            parse_param("1", ParserMode::Strict),
-            Ok(Param {
-                index: Some(1),
-                ty: Type::Format,
-                hint: None,
-            })
+    use rstest::rstest;
+
+    #[rstest]
+    #[case::no__param("", None, Type::Format, None)]
```

Comment:
This gives a warning `function `case_1_no__param` should have a snake case name`.

Reply:
Damn 😑

Reply:
It's solved now